src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/finallocalvariable/InputFinalLocalVariableCheckSwitchAssignment.java

```diff
@@ -0,0 +1,59 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.finallocalvariable;
+
+/* Config:
+ * default
+ */
+public class InputFinalLocalVariableCheckSwitchAssignment {
+    private static final int staticValue = 2;
+    private static final int staticField = switch(staticValue) {
+        case 0 -> -1;
+        case 2-> 2;
+        default -> 3;
+    };
+
+    public InputFinalLocalVariableCheckSwitchAssignment() throws Exception {
+        int a = 0; // violation
+        final int b = 0;
+        int c = 10;
+        int d = 0;
+
+        c = switch (a) {
+            case 0 -> 5;
+            case 1 -> 10;
+            default -> switch(b) {
+                    case 2 -> {
+                        if (a == 0) {
+                            d = 1; // reassign d
+                        }
+                        throw new Exception();
+                    }
+                    default -> 2;
+                };
+        };
+    }
+
+    public void foo() {
+        final int a = 0;
+
+        int b = switch(a) { // violation
```

Comment:
Please add test case for violation inside `case`

Reply:
fixed